students/navdeep/session06/mailroom_part4.py

```diff
@@ -0,0 +1,149 @@
+    user_option = ""
+    while user_option.lower() != 'e':
+        displayMenu()
+        user_option = input("Select an option from the menu: ")
+        if user_option.lower() == 't':
+            addNewDonation(donor_list)
+        elif user_option.lower() == 'r':
+            createReport(donor_list)
+        elif user_option.lower() == 's':
+            writeToFile(donor_list)
+        elif user_option.lower() == 'e':
+            print("Thank you for all of the donations!")
+        else:
+            print("That is not a valid option you chose")
+
+def displayMenu():
+    """
+    displays the menu of options to the user
+    :return:
+    """
+    print()
+    str_menu = "1. Type 't' to Send a Thank You to add a name, donation and thank you card\n"  \
+               "2. Type 'r' to Create a Report and see a list of the donors\n" \
+               "3. Type 's' to Send letters to everyone\n" \
+               "4. Type 'e' to exit the program"
+    print(str_menu)
+    print()
+
+def addNewDonation(donor_list):
+    """
+    Allows user to add a new donor and their donation to the list
+    :param donor_list: the list of donors and their donations
+    """
+    new_donor = input("1. Enter the name of a new donor\n"
+                      "2. Type 'list' to list the donor names\n"
+                      "3. Type 'exit' to return to main menu\n"
+                      "Your Selection:  ")
+    print()
+    if new_donor.lower() == 'list':
+        showDonorNames(donor_list)
+    elif new_donor.lower() == 'exit':
+        return
+    else:
+        try:
+            new_donation = float(input("Enter a new donation amount: "))
+        except ValueError as e:
+            print("The amount you entered is an invalid value.")
+        else:
+            while new_donation < 1:
+                new_donation = float(input("Enter something greater than 0: "))
+            if new_donor.strip() in donor_list:
+                donor_list[new_donor].append(new_donation)
+            else:
+                donor_list[new_donor.strip()] = [new_donation]
+            printThankYou(new_donor, new_donation)
+
+def showDonorNames(donor_list):
+    """
+    will show the unique names of donors.  Will use comprehension to print
+    out the donor names
+    :param donor_list: the list of donors and their donations
+    """
+    print("Here are the list of donors: ")
+    [print(donor) for donor in donor_list.keys()]
+    #for donor in donor_list.keys():
+     #   print(donor)
+
+def printThankYou(newest_donor, newest_donation):
+    """
+    Prints out a thank you message after someone has donated
+    :param newest_donor: the name of the newest donor
+    :param newest_donation: the amount the new donor donated
+    """
+    thank_you_message = "Thank you {} for you generous donation " \
+                        "of ${:.2f}.".format(newest_donor.title(), round(newest_donation,2))
+    print(thank_you_message)
+    print()
```

Comment:
You can also add an "end" argument to the print statement for more control over the output.